app/invocation/invocation_action_card.tsx

```diff
@@ -319,122 +319,161 @@ export default class InvocationActionCardComponent extends React.Component<Props
     );
   }
 
-  private renderTimelines() {
-    const metadata = this.state.actionResult?.executionMetadata;
-
+  private renderTimelines(metadata: build.bazel.remote.execution.v2.ExecutedActionMetadata) {
     type TimelineEvent = {
       name: string;
       color: string;
-      timestamp?: Timestamp | null;
+      timestamp: Timestamp | null | undefined;
     };
     const events: TimelineEvent[] = [
       {
         name: "Queued",
         color: "#3F51B5",
-        timestamp: metadata?.queuedTimestamp,
+        timestamp: metadata.queuedTimestamp,
       },
       {
         name: "Initializing",
         color: "#673AB7",
-        timestamp: metadata?.workerStartTimestamp,
+        timestamp: metadata.workerStartTimestamp,
       },
       {
         name: "Downloading inputs",
         color: "#FF6F00",
-        timestamp: metadata?.inputFetchStartTimestamp,
+        timestamp: metadata.inputFetchStartTimestamp,
       },
       {
         name: "Preparing runner",
         color: "#673AB7",
-        timestamp: metadata?.inputFetchCompletedTimestamp,
+        timestamp: metadata.inputFetchCompletedTimestamp,
       },
       {
         name: "Executing",
         color: "#1E88E5",
-        timestamp: metadata?.executionStartTimestamp,
+        timestamp: metadata.executionStartTimestamp,
       },
       {
         name: "Preparing for upload",
         color: "#673AB7",
-        timestamp: metadata?.executionCompletedTimestamp,
+        timestamp: metadata.executionCompletedTimestamp,
       },
       {
         name: "Uploading outputs",
         color: "#FF6F00",
-        timestamp: metadata?.outputUploadStartTimestamp,
+        timestamp: metadata.outputUploadStartTimestamp,
       },
       // End marker -- not actually rendered.
       {
         name: "Upload complete",
         color: "",
-        timestamp: metadata?.outputUploadCompletedTimestamp,
+        timestamp: metadata.outputUploadCompletedTimestamp,
       },
     ];
 
-    const filteredEvents = events.filter((event) => event && event.timestamp);
-    if (filteredEvents.length == 0) {
+    type FilteredTimelineEvent = {
+      name: string;
+      color: string;
+      timestamp: Timestamp;
+    };
+    let filteredEvents: FilteredTimelineEvent[] = events.filter(
+      (event): event is FilteredTimelineEvent => event.timestamp != null
+    );
+
+    if (filteredEvents.length <= 1) {
```

Comment:
nit, why the change from `== 0` to `<= 1`? seems like if we have 1 event we could render it without any errors

Reply:
I don't think I did it intentionally. Probably was testing out something with the mock data. Will revert.